execution/block-partitioner/src/lib.rs

```diff
@@ -2,61 +2,60 @@
 // Parts of the project are originally copyright © Meta Platforms, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::sharded_block_partitioner::ShardedBlockPartitioner;
-use aptos_types::block_executor::partitioner::RoundId;
-
 pub mod sharded_block_partitioner;
+pub mod v2;
+
 pub mod test_utils;
 
-pub struct BlockPartitionerConfig {
-    num_shards: usize,
-    max_partitioning_rounds: RoundId,
-    cross_shard_dep_avoid_threshold: f32,
-    partition_last_round: bool,
+use aptos_types::{
+    block_executor::partitioner::{PartitionedTransactions, ShardId},
+    transaction::analyzed_transaction::{AnalyzedTransaction, StorageLocation},
+};
+use move_core_types::account_address::AccountAddress;
+use sharded_block_partitioner::config::PartitionerV1Config;
+use std::{
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+};
+use v2::config::PartitionerV2Config;
+mod pre_partition;
+
+pub trait BlockPartitioner: Send {
+    fn partition(
+        &self,
+        transactions: Vec<AnalyzedTransaction>,
+        num_shards: usize,
```

Comment:
Other configs like 'num_threads', 'num_rounds_limit', etc are not a part of this trait but rather part of the obj (PartitionerV2) implementing this trait. Why is 'num_shards' a parameter here ? That is, what is special about 'num_shards' vs say 'num_rounds_limit' ?

Reply:
IMO `num_shards ` actually doesn't have to be a parameter here. I think we made the trait too early...

Reply:
Can you fix the trait then, if it is simple ? or else maybe create a low priority tech debt ?
Upto you, I am not particular on either

Reply:
> create a low priority tech debt ?
do you mean create a TODO comment?

Reply:
sure TODO comment is good. You can resolve this comment after that

Reply:
fixed